cmd/snap-update-ns/user.go

```diff
@@ -37,16 +37,24 @@ type UserProfileUpdateContext struct {
 }
 
 // NewUserProfileUpdateContext returns encapsulated information for performing a per-user mount namespace update.
-func NewUserProfileUpdateContext(instanceName string, uid int) *UserProfileUpdateContext {
+func NewUserProfileUpdateContext(instanceName string, fromSnapConfine bool, uid int) *UserProfileUpdateContext {
 	return &UserProfileUpdateContext{
 		CommonProfileUpdateContext: CommonProfileUpdateContext{
 			instanceName:       instanceName,
+			fromSnapConfine:    fromSnapConfine,
 			desiredProfilePath: desiredUserProfilePath(instanceName),
 		},
 		uid: uid,
 	}
 }
 
+// Lock acquires locks / freezes needed to synchronize mount namespace changes.
+func (ctx *UserProfileUpdateContext) Lock() (unlock func(), err error) {
+	// TODO: when persistent user mount namespaces are enabled, grab a lock
+	// protecting the snap and freeze snap processes here.
+	return func() {}, nil
```

Comment:
why like this instead of the usual Lock/Unlock pair? this seems weird to me

Reply:
Because this way the unlock function holds the reference to a flock-based lock. Otherwise the update context would have to hold more state. I don't really mind either way.